src/mem/pagealloc/bitmask.c

```diff
@@ -112,7 +112,7 @@ struct page_allocator *page_allocator_init(char *start, size_t len, size_t page_
 	unsigned int pages;
 	size_t bitmap_len;
 
-	if (len < page_size) {
+	if (len < page_size + 1) {
```

Comment:
As I can see page_allocator_init() doesn't assume len % page_size == 0. That means, even after your fix if you pass len = page_size + 2 into this function you will get the assertion failure anyway.

My proposition is to change assert(pages>0) to return NULL. In that case this function will handle all possible values of 'len' without assertion failures.

Another possible way to make this function correct is to force 'len % page_size == 0'. I mean assert(len % page_size == 0); at the beginning.